data/lib/compat/compat.lua

```diff
@@ -132,6 +132,42 @@ do
 	rawgetmetatable("Action").__newindex = ActionNewIndex
 end
 
+do
+	local function NpcTypeNewIndex(self, key, value)
+		if key == "onSay" then
+			self:eventType("say")
+			self:onSay(value)
+			return
+		elseif key == "onDisappear" then
+			self:eventType("disappear")
+			self:onDisappear(value)
+			return
+		elseif key == "onAppear" then
+			self:eventType("appear")
+			self:onAppear(value)
+			return
+		elseif key == "onMove" then
+			self:eventType("move")
+			self:onMove(value)
+			return
+		elseif key == "onPlayerCloseChannel" then
+			self:eventType("closechannel")
+			self:onPlayerCloseChannel(value)
+			return
+		elseif key == "onPlayerEndTrade" then
+			self:eventType("endtrade")
+			self:onPlayerEndTrade(value)
+			return
+		elseif key == "onThink" then
+			self:eventType("think")
+			self:onThink(value)
+			return
+		end
```

Comment:
How about we put these in a table rather than if/elseif chaining?

Reply:
I would do this in another pr and include all of the others ones in the similar style then, so we change all of them at once in one pr instead ok?